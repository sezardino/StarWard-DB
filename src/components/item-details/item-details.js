import React from 'react';
import PropTypes from 'prop-types';

import './item-details.css';

const ItemDetails = (props) => {
	const {data, children} = props;
	if (!data) {
		return <p>Choose a character</p>;
	}
	const {name, img} = data;

	return (
		<div className="person-details card">
			<img className="person-image" src={img} alt={name} />

			<div className="card-body">
				<h4>{name}</h4>
				<ul className="list-group list-group-flush">
					{React.Children.map(children, (child) => React.cloneElement(child, {data}))}
				</ul>
			</div>
		</div>
	);
};

const Record = (props) => {
	const {label, field, data} = props;
	return (
		<li className="list-group-item">
			<span className="term">{label}</span>
			<span>{data[field]}</span>
		</li>
	);
};

ItemDetails.propTypes = {
	data: PropTypes.object,
	children: PropTypes.node.isRequired,
};

Record.propTypes = {
	label: PropTypes.string.isRequired,
	field: PropTypes.string.isRequired,
	data: PropTypes.object,
};

export {Record, ItemDetails};
