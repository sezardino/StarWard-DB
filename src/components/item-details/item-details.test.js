import React from 'react';
import {mount} from 'enzyme';

import {ItemDetails, Record} from './item-details';

describe('Component ItemDetails', () => {
	const testData = {
		id: '1',
		name: 'Test name',
		gender: 'Test gender',
		img: 'Test img',
		someData: 'Test data',
	};
	const component = mount(
		<ItemDetails data={testData}>
			<Record field="gender" label="gender label" data={testData} />
			<Record field="someData" label="Some label" data={testData} />
		</ItemDetails>
	);
	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});
});
