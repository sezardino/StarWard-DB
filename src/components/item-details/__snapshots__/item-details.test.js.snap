// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ItemDetails Component rendered correctly 1`] = `
"<ItemDetails data={{...}}>
  <div className=\\"person-details card\\">
    <img className=\\"person-image\\" src=\\"Test img\\" alt=\\"Test name\\" />
    <div className=\\"card-body\\">
      <h4>
        Test name
      </h4>
      <ul className=\\"list-group list-group-flush\\">
        <Record field=\\"gender\\" label=\\"gender label\\" data={{...}}>
          <li className=\\"list-group-item\\">
            <span className=\\"term\\">
              gender label
            </span>
            <span>
              Test gender
            </span>
          </li>
        </Record>
        <Record field=\\"someData\\" label=\\"Some label\\" data={{...}}>
          <li className=\\"list-group-item\\">
            <span className=\\"term\\">
              Some label
            </span>
            <span>
              Test data
            </span>
          </li>
        </Record>
      </ul>
    </div>
  </div>
</ItemDetails>"
`;
